src/mlpack/methods/ann/layer/max_pooling_impl.hpp

```diff
@@ -159,6 +216,47 @@ void MaxPooling<InputDataType, OutputDataType>::serialize(
   ar & BOOST_SERIALIZATION_NVP(outputHeight);
 }
 
+template<typename InputDataType, typename OutputDataType>
+void MaxPooling<InputDataType, OutputDataType>::CreatePadding(
```

Comment:
Just a suggest, might not be necessary, Since across all files (CNN, T-CNN etc) same and valid padding are called from constructor rather than a function, could you implement in a similar way.
Thanks.